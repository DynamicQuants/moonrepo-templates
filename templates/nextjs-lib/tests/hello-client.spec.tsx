import { cleanup, render, screen } from '@testing-library/react';

import HelloClient from '../src/lib/hello-client';

describe('HelloClient', () => {
  beforeEach(() => {
    render(<HelloClient />);
  });

  afterEach(() => {
    cleanup();
  });

  it('should match the snapshot', () => {
    const { container } = render(<HelloClient />);
    expect(container).toMatchSnapshot();
  });

  it('should have h1 with correct text', () => {
    const heading = screen.getByRole('heading', { name: 'Hello from Client!' });
    expect(heading).toBeInTheDocument();
  });
});
