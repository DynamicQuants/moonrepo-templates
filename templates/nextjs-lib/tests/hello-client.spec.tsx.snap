// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloClient should match the snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Hello from Client!
    </h1>
  </div>
</div>
`;
